### DISCUSSION TITLE
How can I get profile picture from auth providers?

### DISCUSSION THREAD

**Discussion by sndrgb** [strapi/strapi#5885] [Legacy Topics]
Hi, I'm not expert to handle oAuth requests, but strapi saved me a big headache - auth with facebook in a couple of hours.

Now, I want to know how can I get additional infos from facebook login, for example the profile picture. I need to import fbsdk in my frontend app? But where I can find facebook id of the user?

Thanks to all

**Comment by lauriejim** [comment]
Hello @sndrgb !
As mentioned [in the docs](https://strapi.io/documentation/3.0.0-beta.x/plugins/users-permissions.html#providers) Strapi use Grant and Purest to connect providers.
[Here](https://github.com/strapi/strapi/blob/37dbd955e2afa6bcb8435ce05897123f71e79f5e/packages/strapi-plugin-users-permissions/services/Providers.js#L117) is the code that manages the data you receive from the provider.
By using the [customization](https://strapi.io/documentation/3.0.0-beta.x/concepts/customization.html#plugin-extensions) concept, you will be able to customize this function to add the data you want. According to the Purest documentation.

**Reply by kevinvugts** [reply]
How would we be able to upload the image?

I tried the following and it didnt work:


```
const files = await fetch(`${profile.avatar}`).then((res) =>
          res.buffer()
        );

        console.log("FILES", files);

        form.append("files", files);
        form.append("ref", "user");
        form.append("refId", createdUser.id);
        form.append("field", "avatar");
        form.append("source", "users-permissions");

        const userJwt = await axios
          .get(
            `${process.env.SERVER_URL}auth/${provider}/callback?access_token=${access_token}`
          )
          .then((res) => res.data);

        await axios
          .post(`${process.env.SERVER_URL}upload`, form, {
            headers: {
              Authorization: `Bearer ${userJwt.jwt}`,
              ...form.getHeaders(),
            },
            cancelToken: source.token,
          })
          .then((res) => console.log("NEW PROFILE PICTURE UPLOAD"))
          .catch((error) => {
            if (error.response) {
              console.log("err", JSON.stringify(error.response.data));
            }
            if (axios.isCancel(error)) {
              console.log("USER CANCELLED THE REQUEST BEFORE COMPLETED");
            }

            console.log("ERROR", JSON.stringify(error));
          });


```

**Reply by derrickmehaffy** [reply]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Comment by jonathanmach** [comment]
Hi, @sndrgb , did you manage to get it working?
I've been stuck trying to implement the same thing.

**Comment by chrissnoek** [comment]
I'm also trying to get this working, but I don't find the documentation really helpful.
Also would like to create a slug out of the FB Username, for example.
Some more extensive help would be appreciated..

**Reply by derrickmehaffy** [reply]
Hi @chrissnoek here you can see what data we are requesting from [facebook](https://github.com/strapi/strapi/blob/37dbd955e2afa6bcb8435ce05897123f71e79f5e/packages/strapi-plugin-users-permissions/services/Providers.js#L176) and [here](https://github.com/strapi/strapi/blob/37dbd955e2afa6bcb8435ce05897123f71e79f5e/packages/strapi-plugin-users-permissions/services/Providers.js#L181-L185) is where we use it, you will want to take a look at the facebook API to see what fields are able to be grabbed from the facebook API. 

As @lauriejim mentioned you will want to use the [extensions system](https://strapi.io/documentation/3.0.0-beta.x/concepts/customization.html#plugin-extensions) to customize the default provider file I listed above to change how data is saved.
